content/client-lib-development-guide/features.textile

```diff
@@ -287,9 +292,10 @@ h3(#realtime-connection). Connection
 * @(RTN6)@ A @Connection@ is successful and considered @CONNECTED@ once the websocket connection is open and the intial @CONNECTED@ @ProtocolMessage@ has been received
 * @(RTN19)@ If the @CONNECTED@ @ProtocolMessage@ contains a @connectionDetails@ property, the attributes within @ConnectionDetails@ will be used as the defaults for this client library, overriding any configured options at the time the @CONNECTED@ @ProtocolMessage@ is received
 * @(RTN7)@ @ACK@ and @NACK@:
-** @(RTN7a)@ All @ProtocolMessage@ @Presence@ and @Message@ objects sent to Ably expect either an @ACK@ or @NACK@ from Ably to confirm successful receipt and acceptance or failure respectively
+** @(RTN7a)@ All @ProtocolMessage@ @Presence@ and @Message@ objects sent to Ably expect either an @ACK@ or @NACK@ from Ably to confirm successful receipt and acceptance or failure respectively. For clarity, it is unnecessary to fail the publish operation of a message using a timer. Instead the client library can rely on the realtime system to send an @ACK@ or @NACK@ when connected, to fail all awaiting messages when @DISCONNECTED@ for too long as described in "RTN7d":#RTN7d, or upon reconnecting, the corresponding @ACK@ or @NACK@ protocol messages will be received as described in "RTN19a":#RTN19a
 ** @(RTN7b)@ Every @ProtocolMessage@ sent must contain a unique serially incrementing @msgSerial@ integer value starting at zero. The @msgSerial@ along with the @count@ for incoming @ACK@ and @NACK@ @ProtocolMessages@ indicates which messages succeeded or failed to be delivered
 ** @(RTN7c)@ If a connection enters the @CLOSED@ or @FAILED@ state, or if the connection state is lost, and a @ACK@ or @NACK@ has not yet been received for a message, the client should consider the delivery of those messages as failed
+** @(RTN7d)@ If a connection enters the @DISCONNECTED@ or @SUSPENDED@ state, and has not become @CONNECTED@ within 15 seconds, then the client should consider the delivery of those messages as failed
```

Comment:
> I didn't see the harm in including the state

I agree.

> he reason I chose 15 is because if we are unable to deliver in that time, I think it's fair that the developer knows about it. 15s for a realtime system is a long time i.e. if I was building a chat app would I not want to know that my message has not arrived in that time?

I definitely agree that it makes no sense to wait for `ConnectionStateTtl` to fail the messages because we could choose arbitrarily large values for that (eg up to 20 mins).

A single momentary disconnection (where the first zero-wait connection attempt succeeds) would permit the messages to be re-sent within the 15s. To get the next 15s attempt to work then this timeout would need to increase to 25s say, which seems a bit long. I think 15s is probably a good number, and if we want to ensure that we get continuity on the second connection attempt (ie the one after the zero-wait attempt) then the right way to do that would be to have an escalating disconnect timeout (eg 5s, 10s, 15s, 15s, ....).